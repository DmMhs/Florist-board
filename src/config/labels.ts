const labels: any = {
  en: {
    brand: 'FLORIST.UA',
    navigation: {
      home: 'HOME',
      shop: 'SHOP',
      gallery: 'GALLERY',
      contacts: 'CONTACTS',
      cart: 'CART',
      account: {
        main: 'MY ACCOUNT',
        menu: {
          signIn: 'SIGN IN',
          signUp: 'SIGN UP',
          logOut: 'LOG OUT'
        }
      },
      lang: 'LANG'
    },
    pages: {
      shop: {
        notAvailable: 'not available :(',
        sort: {
          main: 'sort by',
          btn: {
            byName: 'NAME',
            byPrice: 'PRICE'
          }
        },
        filter: {
          main: 'Filters',
          available: 'IN STOCK',
          price: 'PRICE RANGE',
          priceInputs: {
            from: 'from',
            to: 'to'
          }
        },
        cart: {
          empty: 'cart is empty',
          btn: "LET'S FIX IT",
          title: 'PRODUCT',
          amount: 'AMOUNT',
          price: 'PRICE',
          total: 'Total Sum:',
          orderBtn: 'ORDER'
        }
      },
      gallery: {
        main: 'Our Gallery'
      },
      contacts: {
        map: 'Where to find us?',
        info: 'Feel free to get in touch with us!'
      },
      auth: {
        email: 'Email:',
        password: 'Password:',
        btn: 'SUBMIT',
        alternative: 'OR',
        isAuth: {
          main: 'You are authenticated',
          sub: 'You need to log out if you want to reathenticate!'
        },
        signup: {
          google: 'Sign Up with a Google:',
          facebook: 'Sign Up with a Facebook:',
          account: 'Already have an account ?',
          btn: 'SIGN IN'
        },
        signin: {
          google: 'Sign In with a Google:',
          facebook: 'Sign In with a Facebook:',
          account: 'Have no account ?',
          btn: 'SIGN UP'
        }
      },
      pageNotFound: 'Page not found',
      productDetails: {
        description: 'Description',
        only: 'only',
        notAvailable: 'not available',
        facebookShare: 'SHARE',
        goShopping: 'GO SHOPPING'
      }
    },
    footer: 'FLORIST.UA © All Rights Reserved'
  },
  uk: {
    brand: 'FLORIST.UA',
    navigation: {
      home: 'ГОЛОВНА',
      shop: 'МАГАЗИН',
      gallery: 'ГАЛЕРЕЯ',
      contacts: 'КОНТАКТИ',
      cart: 'КОРЗИНА',
      account: {
        main: 'МІЙ АККАУНТ',
        menu: {
          signIn: 'УВІЙТИ',
          signUp: 'РЕЄСТРАЦІЯ',
          logOut: 'ВИЙТИ'
        }
      },
      lang: 'МОВА'
    },
    pages: {
      shop: {
        notAvailable: 'немає :(',
        sort: {
          main: 'сортувати за',
          btn: {
            byName: "IМ'ЯМ",
            byPrice: 'ЦІНОЮ'
          }
        },
        filter: {
          main: 'Фільтри',
          available: 'В НАЯВНОСТІ',
          price: 'ЦІНОВИЙ ДІАПАЗОН',
          priceInputs: {
            from: 'від',
            to: 'до'
          }
        },
        cart: {
          empty: 'кошик порожній',
          btn: 'ВИПРАВИМО ЦЕ',
          title: 'ПРОДУКТ',
          amount: 'КІЛЬКІСТЬ',
          price: 'ЦІНА',
          total: 'Загальна сума:',
          orderBtn: 'ЗАМОВИТИ'
        }
      },
      gallery: {
        main: 'Наша Галерея'
      },
      contacts: {
        map: 'Де нас можна знайти?',
        info: "Зв'яжіться з нами - ми вам допоможемо!"
      },
      auth: {
        email: 'Е-почта:',
        password: 'Пароль:',
        btn: 'ПІДТВЕРДИТИ',
        alternative: 'АБО',
        isAuth: {
          main: 'Ви вже авторизовані',
          sub: 'Для повторної авторизації вийдіть зі свого аккаунту!'
        },
        signup: {
          google: 'Увійти з Google:',
          facebook: 'Увійти з Facebook:',
          account: 'Вже маєте аккаунт?',
          btn: 'УВІЙТИ'
        },
        signin: {
          google: 'Увійти з Google:',
          facebook: 'Увійти з Facebook:',
          account: 'Ще не зареєстровані?',
          btn: 'РЕЄСТРАЦІЯ'
        }
      },
      pageNotFound: 'Сторінку не знайдено',
      productDetails: {
        description: 'Опис',
        only: 'тільки',
        notAvailable: 'немає в наявності',
        facebookShare: 'ПОДІЛИТИСЬ',
        goShopping: 'В МАГАЗИН'
      }
    },
    footer: 'FLORIST.UA © Всі Права Захищено'
  }
};

export default labels;
