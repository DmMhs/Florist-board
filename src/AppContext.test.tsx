import React from 'react';
import { shallow, mount } from 'enzyme';

import AppContextProvider from './AppContext';
import { BrowserRouter } from 'react-router-dom';
import labels from './config/labels';

describe('AppContext works as expected', () => {
  it('AppContext matches a snapshot', () => {
    const wrapper = shallow(<AppContextProvider />);
    expect(wrapper).toMatchSnapshot();
  });

  it('setLangHandler changes the lang', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider />
      </BrowserRouter>
    );
    const instance = wrapper.find('AppContextProvider').instance();
    instance.setState({
      lang: 'en'
    });
    instance.setLangHandler('uk');
    expect(instance.state.lang).toEqual('uk');
  });

  it('Setting user credentails and logout change Auth context state', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider.WrappedComponent />
      </BrowserRouter>
    );
    const instance = wrapper.find('AppContextProvider').instance();
    instance.setState({
      userLogin: '',
      userId: '',
      userToken: '',
      userAuthenticated: false,
      authenticationMethod: undefined,
      userRole: '',
      lang: 'en',
      labels: {},
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    instance.setUserCredentialsHandler(
      'testLogin',
      'testId',
      'testToken',
      '',
      ''
    );
    expect(instance.state).toEqual({
      userLogin: 'testLogin',
      userId: 'testId',
      userToken: 'testToken',
      userAuthenticated: true,
      authenticationMethod: '',
      userRole: '',
      lang: 'en',
      labels: {},
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    instance.logoutHandler();
    expect(instance.state).toEqual({
      userLogin: '',
      userId: '',
      userToken: '',
      userAuthenticated: false,
      authenticationMethod: undefined,
      userRole: '',
      lang: 'en',
      labels: {},
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
  });

  it('Changes state if local storage auth credentials are not empty', () => {
    localStorage.floristAuthToken = 'asfasfasf';
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider.WrappedComponent />
      </BrowserRouter>
    );
    const instance = wrapper.find('AppContextProvider').instance();
    expect(instance.state.userAuthenticated).toBeTruthy();
  });
});
