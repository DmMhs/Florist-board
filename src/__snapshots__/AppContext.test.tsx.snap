// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContext works as expected AppContext matches a snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
